lib/CoffeeTags/parser.rb

```diff
@@ -112,12 +126,46 @@ def item_for_regex line,  regex, level, additional_fields={}
       end
     end
 
+    # get rid of duplicate entries
+    #def uniq_tree tree
+    #  # group by name first
+    #  groups = tree.group_by {|o| o[:name]}
+    #  tree
+    #end
+
+    # Look through the tree to see if it's possible to change an entry's kind
+    # Differ the objects from simple variables
+    # Should execute after the whole tree has been generated
+    def comb_kinds tree
+      entries_with_parent = tree.reject {|c| c[:parent].nil? }
+      tree.each do |c|
+        next c unless c[:kind] == 'v'
+        maybe_child = entries_with_parent.select {|e| e[:parent] == c[:name]}
+        unless maybe_child.empty?
+          c[:kind] = 'o'
+        end
+      end
+      tree
+    end
+
+    # trim the bloated tree
+    # - when not required to include_vars, reject the variables
+    def trim_tree tree
+      unless @include_vars
+        tree = tree.reject do |c|
+          ['v'].include? c[:kind]
```

Comment:
It might be block or function...etc,  but  this method's purpose is to find objects that be identified as simple variable